doc/api/process.md

```diff
@@ -1996,6 +1996,129 @@ Will generate an object similar to:
   tz: '2016b' }
 ```
 
+## Access Control
+
+> Stability: 1 - Experimental
+
+### process.accessControl.apply(restrictions)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `restrictions` {Object} A set of access restrictions that will be applied
+  to the current Node.js instance. The format should be the same as the one
+  returned by [`process.accessControl.getCurrent()`][]. Omitted keys will
+  default to `true`, i.e. to retaining the relevant permissions.
+
+*Warning*: This does not provide a full isolation mechanism. Child processes,
+native addons, and other communication channels may be used to circumvent
+these measures, if they have been made available before the corresponding
+restrictions have been put into place.
+
+This API is recent and may not be complete.
+Please report bugs at https://github.com/nodejs/node/issues.
+
+Operations scheduled before this call are not undone or stopped.
+Features that were previously disabled through this call cannot be re-enabled.
+Child processes do not inherit these restrictions, whereas [`Worker`][]s do.
+
+The following code removes some permissions of the current Node.js instance:
+
+```js
+process.accessControl.apply({
+  childProcesses: false,
+  createWorkers: false,
+  fsRead: false,
+  fsWrite: false,
+  loadAddons: false,
+  setV8Flags: false
+});
+```
+
+Keys not listed in the object, or with values not set to `false`,
+are unaffected.
+
+See [`process.accessControl.getCurrent()`][] for a list of permissions.
+
+### process.accessControl.getCurrent()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Object} An object representing a set of permissions for the
+  current Node.js instance of the following structure:
+
+<!-- eslint-skip -->
+```js
+{ accessInspectorBindings: true,
+  childProcesses: true,
+  createWorkers: true,
+  fsRead: true,
+  fsWrite: true,
+  loadAddons: true,
+  modifyTraceEvents: true,
+  netConnectionless: true,
+  netIncoming: true,
+  netOutgoing: true,
+  setEnvironmentVariables: true,
+  setProcessAttrs: true,
+  setV8Flags: true,
+  signalProcesses: true }
+```
+
+Currently only boolean options are supported.
+
+In particular, these settings affect the following features of the Node.js API:
+
+- `accessInspectorBindings`:
+  - [`inspector.open()`][]
+  - [`inspector.Session.connect()`][]
+- `childProcesses`:
+  - [`child_process`][`ChildProcess`] methods
+- `createWorkers`:
+  - [`worker_threads.Worker`][`Worker`]
+- `fsRead`:
+  - All read-only [`fs`][] operations, including watchers
+  - [`os.homedir()`][]
+  - [`os.userInfo()`][]
+  - [`require()`][]
+  - Note that lack of `fsRead` permission does not imply lack of `fsWrite`
+    permission. In particular, [`fs.write()`][] calls on already opened file
+    descriptors will succeed.
+- `fsWrite`:
```

Comment:
Do you think it would be worthwhile to remark on `open` explicitly, since it's a vectored API whose behavior depends on the flags (and the access check looks at the flags)?

Reply:
Yes, makes sense. Done!